lib/Backend/Vault/Kv.hs

```diff
@@ -87,14 +88,24 @@ data CofferSpecials =
   deriving anyclass (A.ToJSON, A.FromJSON)
 makeLensesWith abbreviatedFields ''CofferSpecials
 
+getManager :: Effects r => ManagerSettings -> Sem r Manager
+getManager settings = do
+  managerMb <- get @(Maybe Manager)
+  case managerMb of
+    Nothing -> do
+      manager <- embed $ newManager settings
+      put (Just manager)
+      pure manager
+    Just manager -> pure manager
+
 getEnv :: Effects r => VaultKvBackend -> Sem r ClientEnv
 getEnv backend =
   case url of
     (BaseUrl Http _ _ _) -> do
-      manager <- embed $ newManager defaultManagerSettings
+      manager <- getManager defaultManagerSettings
       pure $ mkClientEnv manager url
     (BaseUrl Https _ _ _) -> do
-      manager <- embed $ newManager tlsManagerSettings
+      manager <- getManager tlsManagerSettings
       pure $ mkClientEnv manager url
```

Comment:
Say we have 2 backends in the config, `b1` uses HTTP and `b2` uses HTTPS.

If we make a request to `b1` first, `getEnv` and `getManager` will create a manager with `defaultManagerSettings`.
If we then make a request to `b2`, `getEnv` and `getManager` will reuse the previous manager, and fail with `TlsNotSupported` (I think).

We need to have (at most) 2 managers, one for HTTP and another for HTTPS requests.

Reply:
Done